File: frontend/src/services/resultService.js
import api from './api'

export const resultService = {
  async getResults(role) {
    const response = await api.get('/results', { params: { role } })
    return response.data
  },

  async getResultById(id) {
    const response = await api.get(`/results/${id}`)
    return response.data
  },

  async createResult(resultData) {
    const response = await api.post('/results', resultData)
    return response.data
  },

  async updateResult(id, resultData) {
    const response = await api.put(`/results/${id}`, resultData)
    return response.data
  },

  async deleteResult(id) {
    const response = await api.delete(`/results/${id}`)
    return response.data
  },

  async getResultsByStudent(studentId) {
    const response = await api.get(`/results/student/${studentId}`)
    return response.data
  },

  async getResultsByExam(examId) {
    const response = await api.get(`/results/exam/${examId}`)
    return response.data
  },

  async getResultsByCourse(courseId) {
    const response = await api.get(`/results/course/${courseId}`)
    return response.data
  },
}

File: backend/src/main/java/com/example/exammanagement/service/ResultService.java
package com.example.exammanagement.service;

import com.example.exammanagement.entity.Result;
import com.example.exammanagement.repository.ResultRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ResultService {
    
    @Autowired
    private ResultRepository resultRepository;
    
    public List<Result> getAllResults() {
        return resultRepository.findAll();
    }
    
    public Result getResultById(Long id) {
        return resultRepository.findById(id)
            .orElseThrow(() -> new RuntimeException("Result not found with id: " + id));
    }
    
    public List<Result> getResultsByStudent(Long studentId) {
        return resultRepository.findByStudentId(studentId);
    }
    
    public List<Result> getResultsByExam(Long examId) {
        return resultRepository.findByExamId(examId);
    }
    
    public Result createResult(Result result) {
        return resultRepository.save(result);
    }
    
    public Result updateResult(Long id, Result resultDetails) {
        Result result = getResultById(id);
        
        result.setMarksObtained(resultDetails.getMarksObtained());
        result.setTotalMarks(resultDetails.getTotalMarks());
        result.setGrade(resultDetails.getGrade());
        result.setRemarks(resultDetails.getRemarks());
        result.setExam(resultDetails.getExam());
        result.setStudent(resultDetails.getStudent());
        
        return resultRepository.save(result);
    }
    
    public void deleteResult(Long id) {
        Result result = getResultById(id);
        resultRepository.delete(result);
    }
}
